### PR TITLE
Rank Rework. [WIP] [DO NOT MERGE]

### PR DESCRIPTION
**Pull request by ActualRory**
**THIS IS A WORK IN PROGRESS. DO NOT MERGE.**

Current Features:
- **Refluffs** various ranks and roles for the Marine Faction.
- **Adds** the role of 'Field Officer'
- **Merges** the roles of 'Cargo Technician' and 'Maintenance Technician.'
- **Merges** Engineering and Cargo radio channels into 'Logistics.'

### CODE REVIEW

File: code/controllers/communications.dm [hunk 1]
```diff
@@ -144,9 +144,8 @@ var/list/radiochannels = list(
 	"Almayer"		= PUB_FREQ,
 	"Command"		= COMM_FREQ,
 	"MedSci"		= MED_FREQ,
-	"Engi"			= ENG_FREQ,
+	"Logistics"		= SUP_FREQ,
```
Comment (Rohesie) [inline code/controllers/communications.dm]:
```suggestion
	"Logistics"		= LOG_FREQ,
```

File: code/game/jobs/access.dm [hunk 2]
```diff
@@ -258,24 +258,26 @@ proc/get_all_job_icons() return joblist + list("Prisoner")//For all existing HUD
 		if("E4") . = size ? "CPL " : "Corporal"
 		if("E5") . = size ? "SGT " : "Sergeant"
 		if("E6") . = size ? "SSGT " : "Staff Sergeant"
-		if("E6E") . = size ? "TSGT " : "Technical Sergeant"
+		if("E6T") . = size ? "TSGT " : "Technical Sergeant"
 		if("E7") . = size ? "SFC " : "Sergeant First Class"
+		if("E7E") . = size ? "GSGT " : "Gunnery Sergeant"
```
Comment (Rohesie) [inline code/game/jobs/access.dm]:
```suggestion
		if("E7E") . = size ? "GySGT " : "Gunnery Sergeant"
```

File: code/game/jobs/access.dm [hunk 3]
```diff
@@ -258,24 +258,26 @@ proc/get_all_job_icons() return joblist + list("Prisoner")//For all existing HUD
...
 		if("E8") . = size ? "MSGT " : "Master Sergeant"
 		if("E8E") . = size ? "FSGT " : "First Sergeant"
 		if("E9") . = size ? "SGM " : "Sergeant Major"
 		if("E9E") . = size ? "CSGM " : "Command Sergeant Major"
 		if("O1") . = size ? "ENS " : "Ensign"
+		if("O1E") . = size ? "2LT " : "Second Lieutenant"
```
Comment (Rohesie) [inline code/game/jobs/access.dm]: There's First Lieutenant (1LT) and Lieutenant Junior Grade (LTJG)

File: code/game/jobs/job/command.dm [hunk 4]
```diff
@@ -92,6 +92,35 @@ Make the USCM proud!"}
...
+/datum/job/command/field
+	title = "Field Officer"
+	comm_title = "FO"
+	paygrade = "O3"
+	flag = ROLE_FIELD_OFFICER
+	flags_startup_parameters = ROLE_ADD_TO_DEFAULT|ROLE_ADD_TO_MODE|ROLE_ADMIN_NOTIFY
+	skills_type = /datum/skills/FO
+
+
+	generate_wearable_equipment()
```
Comment (Rohesie) [inline code/game/jobs/job/command.dm]:
```suggestion
/datum/job/command/field/generate_wearable_equipment()
```

File: code/game/jobs/job/command.dm [hunk 5]
```diff
@@ -92,6 +92,35 @@ Make the USCM proud!"}
...
+				WEAR_EAR = /obj/item/device/radio/headset/almayer/mcom,
+				WEAR_BODY = /obj/item/clothing/under/marine/officer/exec,
+				WEAR_FEET = /obj/item/clothing/shoes/marine,
+				WEAR_WAIST = /obj/item/storage/belt/gun/m4a3/vp70,
+				WEAR_HEAD = /obj/item/clothing/head/cmcap,
+				WEAR_BACK = /obj/item/storage/backpack/marine/satchel,
+				WEAR_R_STORE = /obj/item/storage/pouch/general/large
+				)
+
+	generate_entry_message(mob/living/carbon/human/H)
```
Comment (Rohesie) [inline code/game/jobs/job/command.dm]:
```suggestion
/datum/job/command/field/generate_entry_message(mob/living/carbon/human/H)
```

File: code/game/jobs/job/logistics.dm [hunk 6]
```diff
@@ -73,17 +73,17 @@ A happy ship is a well-functioning ship."}
...
 	flag = ROLE_MAINTENANCE_TECH
 	department_flag = ROLEGROUP_MARINE_ENGINEERING
 	faction = "Station"
-	total_positions = 4
-	spawn_positions = 4
+	total_positions = 6
+	spawn_positions = 6
 	scaled = 1
-	supervisors = "the chief engineer"
+	supervisors = "the chief engineer, the requisition officer"
```
Comment (Rohesie) [inline code/game/jobs/job/logistics.dm]:
```suggestion
	supervisors = "the requisitions officer and the chief engineer"
```

File: code/game/objects/items/devices/radio/headset.dm [hunk 7]
```diff
@@ -274,7 +274,7 @@
 
 /obj/item/device/radio/headset/almayer/ce
 	name = "chief engineer's headset"
-	desc = "The headset of the guy who is in charge of morons. To access the engineering channel, use :e. For command, use :v."
+	desc = "The headset of the guy who is in charge of morons. To access the logistics channel, use :u. For command, use :v."
```
Comment (Rohesie) [inline code/game/objects/items/devices/radio/headset.dm]: Is `:l` taken?

**Review by Rohesie — CHANGES_REQUESTED** [review]

File: code/controllers/communications.dm (hunk 1 above)
Comment (ActualRory) [reply code/controllers/communications.dm]: Did this at first, but pretty sure it breaks some stuff, which is why CM was using `SUP_FREQ` for Requistions. Will look into making the changeover soon.

File: code/game/objects/items/devices/radio/headset.dm (hunk 7 above)
Comment (ActualRory) [reply code/game/objects/items/devices/radio/headset.dm]: Currerntly piggybacking off of `SUP_FREQ` which is tied to U. 

Can switch over to :l after I wade into that.

File: code/game/jobs/access.dm (hunk 3 above)
Comment (ActualRory) [reply code/game/jobs/access.dm]: Is there?
I can't think of a single military that uses LTJG->1LT.
It usually goes one of the following:
- 2LT->LT
- 2LT ->1LT
- SLT (Sub-Lieutenant) -> LT
- LT(JG) -> LT(SG)

Comment (Rohesie) [reply code/game/jobs/access.dm]: The US Navy uses LTJG for this rank, or Sub-Lieutenant. Anything works here, really.

**Pull request closed without merging**